test/system/scenarios/invalid-orders.js

```diff
@@ -0,0 +1,84 @@
+    .then(() => trader2.createOrder({ type: 'L', side: 'A', quantity: 1, price: 0.5}, { expectFail: true, expectFailReason: 'limit order with invalid side' }))
+    .then(() => trader2.createOrder({ type: 1, side: 'B', quantity: 1, price: 0.5}, { expectFail: true,
+      expectFailReason: 'limit order with type as integer' }))
+    .then(() => trader2.createOrder({ type: 'L', side: 'S', quantity: 'g', price: 0.5}, { expectFail: true,
+      expectFailReason: 'limit order with quantity as string' }))
+    .then(() => trader2.createOrder({ type: 'L', side: 'B', price: 0.5}, { expectFail: true, expectFailReason: 'limit order without qty' }))
+    .then(() => trader2.createOrder({ type: 'L', side: 'B', quantity: 0, price: 0.5}, { expectFail: true, expectFailReason: 'limit order with 0 qty' }))
+    .then(() => trader2.createOrder({ type: 'L', side: 'B', quantity: 0.5, price: 0}, { expectFail: true, expectFailReason: 'limit order with 0 price' }))
+    .then(() => trader2.createOrder({ type: 'L', side: 'B', quantity: 0.5}, { expectFail: true, expectFailReason: 'limit order without price' }));
+
+}
+
+function testStopOrder(trader3) {
+  return Promish.resolve()
+    .then(() => trader3.createOrder({type: 'S', side: 'S', quantity: 0.5}, {expectFail: true, expectFailReason: 'stop order without stop value'}))
+    .then(() => trader3.createOrder({ type: 'S', side: 'S', quantity: 0.5, stop: -0.2}, { expectFail: true,
+      expectFailReason: 'stop order with negative stop value' }))
+    .then(() => trader3.createOrder({type: 'S', side: 'S', quantity: 0.5, stop: 0}, {expectFail: true, expectFailReason: 'stop order with 0 as stop value'}))
+    .then(() => trader3.createOrder({type: 'S', side: 'S', quantity: 0.5, stop: 0}, {expectFail: true, expectFailReason: 'stop order with 0 as stop value'}))
+    .then(() => trader3.createOrder({type: 'S', side: 'B', quantity: 0.5, stop: 1}, {expectFail: true, expectFailReason: 'stop buy order without value'}));
+
+}
+function testStopLimitOrder(trader2) {
+  return Promish.resolve()
+    .then(() => trader2.createOrder({ type: 'SL', side: 'B', quantity: 0.5, price:0.25, stop: 0}, { expectFail: true,
+      expectFailReason: 'stop limit order with 0 stop value' }))
+    .then(() => trader2.createOrder({ type: 'SL', side: 'B', quantity: 0.5, price:0.25, stop: -0.5}, { expectFail: true,
+      expectFailReason: 'stop limit with negative stop value' }))
+    .then(() => trader2.createOrder({ type: 'SL', side: 'S', quantity: 0.5, price:0.25, stop: 'b'}, { expectFail: true,
+      expectFailReason: 'stop limit with string stop value' }))
+    .then(() => trader2.createOrder({ type: 'SL', side: 'S', quantity: 0.5, price:0.25 }, { expectFail: true,
+      expectFailReason: 'stop limit order without stop value' }));
+}
+
+function testTrailingStopOrder(trader3) {
+  return Promish.resolve()
```

Comment:
Also test for:
* invalid trail values (zero, negative, strings, etc)
* invalid trailType values (numeric, missing)